src/plugins/intel_cpu/src/config.cpp

```diff
@@ -73,192 +68,248 @@ void Config::applyDebugCapsProperties() {
 void Config::readProperties(const ov::AnyMap& prop, const ModelType modelType) {
     const auto streamExecutorConfigKeys =
         streamExecutorConfig.get_property(ov::supported_properties.name()).as<std::vector<std::string>>();
-    const auto hintsConfigKeys = perfHintsConfig.SupportedKeys();
     for (const auto& kvp : prop) {
         const auto& key = kvp.first;
-        const auto& val = kvp.second.as<std::string>();
-        IE_SUPPRESS_DEPRECATED_START
+        const auto& val = kvp.second;
         if (streamExecutorConfigKeys.end() !=
             std::find(std::begin(streamExecutorConfigKeys), std::end(streamExecutorConfigKeys), key)) {
-            streamExecutorConfig.set_property(key, val);
+            streamExecutorConfig.set_property(key, val.as<std::string>());
             if (key == ov::affinity.name()) {
-                const auto affinity_val = ov::util::from_string(val, ov::affinity);
-                if (affinity_val == ov::Affinity::CORE || affinity_val == ov::Affinity::HYBRID_AWARE) {
-                    enableCpuPinning = true;
-                    changedCpuPinning = true;
-                } else if (affinity_val == ov::Affinity::NUMA) {
-                    enableCpuPinning = false;
-                    changedCpuPinning = true;
+                changedCpuPinning = true;
+                try {
+                    const auto affinity_val = val.as<ov::Affinity>();
+                    enableCpuPinning =
+                        (affinity_val == ov::Affinity::CORE || affinity_val == ov::Affinity::HYBRID_AWARE) ? true
+                                                                                                           : false;
+                } catch (const std::exception&) {
+                    OPENVINO_THROW("Wrong value ",
+                                   val.as<std::string>(),
+                                   "for property key ",
+                                   key,
+                                   ". Expected only CORE/NUMA/HYBRID_AWARE.");
                 }
             }
-        } else if (hintsConfigKeys.end() != std::find(hintsConfigKeys.begin(), hintsConfigKeys.end(), key)) {
-            perfHintsConfig.SetConfig(key, val);
+        } else if (key == ov::hint::performance_mode.name()) {
+            try {
+                hintPerfMode = val.as<ov::hint::PerformanceMode>();
+            } catch (const std::exception&) {
+                OPENVINO_THROW("Wrong value ",
+                               val.as<std::string>(),
+                               "for property key ",
+                               key,
+                               ". Expected only LATENCY/THROUGHPUT/CUMULATIVE_THROUGHPUT.");
+            }
+        } else if (key == ov::hint::num_requests.name()) {
+            try {
+                ov::Any value = val.as<std::string>();
+                int val_i = value.as<int>();
```

Comment:
can we use directly `as<int>` and avoid intermediate conversion to string?

Reply:
It seems there is some problem: `ov::hint::num_requests` is `uint32_t` type, so `as<int>` will always failed:

```
C++ exception with description "Exception from src/core/include/openvino/core/any.hpp:841:
Bad cast from: j to: i
" thrown in the test body.
```

We can remove `if(val_i < 0)` branch and replace it with as`<uint32_t>()`, which will report exception if the val is negative.
So we can only use one line:
```
try {
    hintNumRequests = val.as<uint32_t>();
}
```

But there is also risk if user uses `int `for `ov::hint::num_requests` like this:
```
    int32_t req_num= 4;
    ov::AnyMap config;
    config[ov::hint::num_requests] = req_num;
```
We can require user to use correct data type for each property, but for some integer value, it is easy to use incorrect data type between `int32_t` and `uint32_t`, so I suggest using `std::string` to unify it to avoid strange exception for user.

Reply:
ok